pkg/executor/docker/bidstrategy/semantic/image_platform.go

```diff
@@ -32,26 +33,38 @@ func (s *ImagePlatformBidStrategy) ShouldBid(
 	ctx context.Context,
 	request bidstrategy.BidStrategyRequest,
 ) (bidstrategy.BidStrategyResponse, error) {
-	if request.Job.Spec.Engine != model.EngineDocker {
+	// TODO(forrest): [correctness] this appears to be a bug as we are returning `true` for ShouldBid
+	if request.Job.Spec.EngineDeprecated != model.EngineDocker {
 		return bidstrategy.NewShouldBidResponse(), nil
 	}
+	// TODO(forrest): [correctness] keep same (probably incorrect) behavior as before.
+	if request.Job.Spec.EngineSpec.Type != model.EngineTypeDocker {
+		return bidstrategy.NewShouldBidResponse(), nil
+	}
+
+	// TODO(forrest): [correctness] keep same (probably incorrect) behavior as before.
+	dockerEngine, engineErr := model.DockerEngineFromEngineSpec(request.Job.Spec.EngineSpec)
+	if engineErr != nil {
+		return bidstrategy.NewShouldBidResponse(), engineErr
+	}
```

Comment:
This all looks like a bug as we are returning an affirmative bidding response for image based bidding the job **is not docker**. Thoughts?

Reply:
An affirmative response is more like "don't discount based on this". So we have to have an affirmative response from everyone, or just a single negative response.